### ISSUE TITLE
Suggestion

### ISSUE THREAD

**Issue by xu6148152** [ks3sdk/KSYLiveAndroidSDK#8]
Using MediaRecorder to record video will delay 700ms and size is limited, it has many sucks, so I think using OpenGl and MediaCodec to merge video is a more better solution.

**Comment by tian9246** [comment]
Thanks for your suggestion , We use MediaRecorder as the first way to obtain H264 raw data but not the only way .
Delay is fact and we don't write file to disk actually so size is not limited.
Can you explain why we use openGL? How to obtain camera's frame data in your way?

**Comment by xu6148152** [comment]
Camera has a interface that onPreview()
Using this method can get every frame of camera

**Comment by xu6148152** [comment]
I said limited size is width and height of video is limited, hardware must support it, or    
 it can not work.

**Comment by tian9246** [comment]
Yeah , we have considered that. but mediacodec first become apart of CTS in Android 4.3 , below this the compatibility is not good 
In our plan software codec is preferential ..

**Comment by xu6148152** [comment]
Yeah, software codec can choose FFMPEG, but its efficiency must be care.
